website/src/components/App.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { Component } from 'react'
+import fire from './config/Fire'
+import Login from './views/Admin/Login'
+import Dashboard from './views/Admin/Dashboard'
+
+class App extends Component {
```

Comment:
Why not a functional component like the others?

Reply:
Because, here I'm using react  component class life cycle method and state

Reply:
But you can call all the logic inside the cdm function before returning the component.
This is what you currently have essentially.
```typescript
class Component {
  componentDidMount() {
    // Removed the function that calls this function to simplify
    fire.auth().onAuthStateChanged( ... )
  }
}
```

You can change this to this
```typescript
async function Component() {
  // call the listener before returning
  fire.auth().onAuthStateChanged( ... )

  return (
   ...
  )
}
```

If you still want to use lifecycle methods you can use the `useEffect` hook and others too.
For state there's already the `useState` hook that you're using.

Reply:
While converting component class to stateless functional component with react hook, I'm getting this error : 

JSX element type 'Promise<Element>' is not a constructor function for JSX elements.
  Type 'Promise<Element>' is missing the following properties from type 'Element': type, props, key  TS2605

    14 |     <Switch>
    15 |       <Route exact path='/' >
  > 16 |         <App />
       |         ^
    17 |       </Route>
    18 |       <Route path='/Login'  >
    19 |         <Login />

Code: 
export default async function App() {

  const [user, setUser] = React.useState(null);

  fire.auth().onAuthStateChanged((user) => {
    if (user) {
      setUser(user);
      // session start
    }
    else {
      setUser(null);
    }
  })
  
  return (
    <div>
      {
        user ? <Dashboard /> : <Login />
      }
    </div>
  )

}